src/fixing/mod.rs

```diff
@@ -92,6 +92,7 @@ pub fn run_fixing_loop<'a>(
 
         let from_file_run_context_instance_provider =
             from_file_run_context_instance_provider_factory.create();
+        let node_parent_cache = get_node_parent_cache(&tree);
```

Comment:
Worth noting that this also could possibly be "incrementally re-computed" for the LSP server use case?

Might be worth inspecting how much memory is being eaten up by these `HashMap`'s with every single AST node in them (especially thinking about then keeping them cached between runs by the LSP server)?